system_test_ci/run.py

```diff
@@ -0,0 +1,278 @@
+import click
+import json
+import yaml
+import paramiko
+import pathlib
+import requests
+
+import mlrun.utils
+
+
+logger = mlrun.utils.create_logger(level="debug", name="ci")
+
+
+class SystemTestCIRunner(object):
+    class Constants(object):
+        ssh_username = "iguazio"
+        mlrun_code_path = pathlib.Path("/home/iguazio/")
+        system_tests_env_yaml = pathlib.Path("tests") / "system" / "env.yml"
+        git_url = "https://github.com/mlrun/mlrun.git"
+        provctl_releases = (
+            "https://api.github.com/repos/iguazio/provazio/releases/latest"
+        )
+        provctl_binary_format = "provctl-{release_name}-linux-amd64"
+
+    def __init__(
+        self,
+        mlrun_version: str,
+        mlrun_repo: str,
+        data_cluster_ip: str,
+        data_cluster_ssh_password: str,
+        app_cluster_ssh_password: str,
+        github_access_key: str,
+        mlrun_dbpath: str,
+        v3io_api: str,
+        v3io_username: str,
+        v3io_password: str,
+        v3io_access_key: str,
+    ):
+        self._logger = logger
+        self._mlrun_version = mlrun_version
+        self._mlrun_repo = mlrun_repo
+        self._data_cluster_ip = data_cluster_ip
+        self._data_cluster_ssh_password = data_cluster_ssh_password
+        self._app_cluster_ssh_password = app_cluster_ssh_password
+        self._github_access_key = github_access_key
+
+        self._clean_up_files = []
+
+        self._env_config = {
+            "MLRUN_DBPATH": mlrun_dbpath,
+            "V3IO_API": v3io_api,
+            "V3IO_USERNAME": v3io_username,
+            "V3IO_PASSWORD": v3io_password,
+            "V3IO_ACCESS_KEY": v3io_access_key,
+        }
+
+        self._logger.info("Connecting to data-cluster")
+        self._ssh_client = paramiko.SSHClient()
+        self._ssh_client.set_missing_host_key_policy(paramiko.WarningPolicy)
+        self._ssh_client.connect(
+            data_cluster_ip,
+            username=self.Constants.ssh_username,
+            password=data_cluster_ssh_password,
+        )
+
+    def run(self):
+        mlrun_path = self.Constants.mlrun_code_path / "mlrun"
+        self._prepare_test_env(mlrun_path)
+        provctl = self._download_povctl()
+        self._patch_mlrun(provctl)
+
+        self._run_command("cd", args=[str(mlrun_path), "&&", "make", "test-system"])
+
+    def clean_up(self):
+        self._logger.info("Cleaning up")
+        if len(self._clean_up_files) > 0:
+            self._run_command(f'rm -f {" ".join(self._clean_up_files)}')
```

Comment:
Using one top work directory you can simply remove it with `rm -rf` and you won't need to remember all of the files